src/async_graphql_hyper.rs

```diff
@@ -133,4 +142,26 @@ impl GraphQLResponse {
     };
     self
   }
+
+  /// Sets the `response_headers` for a given `GraphQLResponse`.
+  /// The function modifies the `GraphQLResponse` to set the `response_headers` to the specified `headers` value.
+  /// # Arguments
+  /// * `res` - The GraphQL response whose `response_headers` is to be set.
+  /// * `headers` - The `response_headers` value to be set for `response_headers`.
+  /// # Returns
+  /// * A modified `GraphQLResponse` with updated `response_headers`.
+
+  pub fn set_response_headers(mut self, headers: HeaderMap) -> GraphQLResponse {
+    match self.0 {
+      BatchResponse::Single(ref mut res) => {
+        res.http_headers.extend(headers.clone());
```

Comment:
This clone is unnecessary.

Reply:
headers are being used in batch response as well. So need to clone it.

Reply:
could be done on response instead of GraphQL response